ms2query/ms2library.py

```diff
@@ -0,0 +1,375 @@
+import sqlite3
+import pickle
+from ms2deepscore.models import load_model as load_ms2ds_model
+from ms2deepscore.models import SiameseModel
+from ms2deepscore import MS2DeepScore
+from ms2deepscore.vector_operations import cosine_similarity_matrix
+from gensim.models import Word2Vec
+from tqdm import tqdm
+from spec2vec import SpectrumDocument
+from spec2vec.vector_operations import calc_vector, cosine_similarity_matrix
+from ms2query.ms2query.s2v_functions import \
+    post_process_s2v
+
+# todo Check for _obj.get() for spectrumdocuments and after new release change
+#  to .get()
+
+
+class Ms2Library:
+    def __init__(self,
+                 sqlite_file_location: str,
+                 s2v_model_file_name: str,
+                 ms2ds_model_file_name: str,
+                 pickled_s2v_embeddings_file_name: str,
+                 pickled_ms2ds_embeddings_file_name: str,
+                 **settings):
+        """
+
+        Args:
+        -------
+        sqlite_file_location:
+            The location at which the sqlite_file_is_stored. The file is
+            expected to have 3 tables: tanimoto_scores, inchikeys and
+            spectra_data. If no sqlite file is available paramater is expected
+            to be None.
+        file_name_dict:
+            A dictionary containing the files needed to create a sql database.
+            The dictionary is expected to contain the keys:
+            "npy_file_location", "pickled_file_name" and "csv_file_name". With
+            as value the location of these files.
+            Default = None
+        """
+        # Set default settings
+        self.mass_tolerance = 1.0
+
+        # Change default settings to values given in **settings
+        self._set_settings(settings)
+
+        # todo check if the sqlite file contains the correct tables
+        self.sqlite_file_location = sqlite_file_location
+        self.s2v_model = Word2Vec.load(s2v_model_file_name)
+        self.ms2ds_model = load_ms2ds_model(ms2ds_model_file_name)
+        # loads the library embeddings into memory
+        with open(pickled_s2v_embeddings_file_name, "rb") as \
+                pickled_s2v_embeddings:
+            self.s2v_embeddings = pickle.load(pickled_s2v_embeddings)
+        with open(pickled_ms2ds_embeddings_file_name, "rb") as \
+                pickled_ms2ds_embeddings:
+            self.ms2ds_embeddings = pickle.load(pickled_ms2ds_embeddings)
+
+    def _set_settings(self,
+                      settings: Dict[str, Any]):
+        """Changes default settings to settings
+
+        Attributes specified in settings are expected to have been defined with
+        a default value before calling this function.
+        Args
+        ------
+        settings:
+            Dictionary with as keys the name of the attribute that should be
+            set and as value the value this attribute should have.
+
+        """
+        # Get all attributes to check if the arguments in settings are allowed
```

Comment:
```suggestion
        """
        # Set default settings
        self.mass_tolerance = 1.0

        # Get all attributes to check if the arguments in settings are allowed
```

Reply:
I agree that this is better structured. Initially I had this solution as well. However, Pycharm (and possibly other interpreters) give warnings when later using self.mass_tolerance, because it is not initialized in init. We can still change it to your function and it would work, but it makes it confusing when working with interpreters that give this kind of warning. Let me know what you think is the best solution.

Reply:
OK, good to know. My intuition said it would be "cleaner" to initiate all defaults in an own function (which alter could be quiet a big dictionary ...). But feel to leave it as it is for now!